src/page-weekly-snapshot/WeeklyReportContext.tsx

```diff
@@ -0,0 +1,192 @@
+};
+
+type StateNameAction = {
+  type: "UPDATE_STATE_NAME";
+  payload: string;
+};
+
+type RequestActions = {
+  type: "REQUEST_SHARED_SCENARIOS";
+};
+
+type WeeklyReportDispatch = (action: any) => void;
+
+interface WeeklyReportContext {
+  state: WeeklyReportState;
+  dispatch: WeeklyReportDispatch;
+}
+
+const WeeklyReportContext = React.createContext<
+  WeeklyReportContext | undefined
+>(undefined);
+
+function weeklyReportReducer(
+  state: WeeklyReportState,
+  action: WeeklyReportActions,
+): WeeklyReportState {
+  switch (action.type) {
+    case "REQUEST_SHARED_SCENARIOS":
+      return Object.assign({}, state, { loading: true });
+    case "RECEIVE_SHARED_SCENARIOS":
+      return Object.assign({}, state, {
+        loading: false,
+        sharedScenarios: action.payload,
+      });
+    case "RECEIVE_SCENARIO":
+      return Object.assign({}, state, {
+        scenario: action.payload,
+      });
+    case "UPDATE_STATE_NAME":
+      return Object.assign({}, state, {
+        stateName: action.payload,
+      });
+    case "RECEIVE_FACILITIES":
+      return Object.assign({}, state, {
+        facilities: action.payload,
+      });
+    default:
+      return state;
+  }
+}
+
+export const WeeklyReportProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [state, dispatch] = React.useReducer(weeklyReportReducer, {
+    loading: false,
+    stateName: null,
+    scenario: null,
+    sharedScenarios: [],
+    facilities: [],
+  });
+
+  useEffect(() => {
+    dispatch({ type: "REQUEST_SHARED_SCENARIOS" });
+    getSharedBaselineScenarios().then((scenarios) => {
+      if (scenarios) {
+        dispatch({ type: "RECEIVE_SHARED_SCENARIOS", payload: scenarios });
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (state.stateName) {
+      getSharedScenarioByStateName(state.sharedScenarios, state.stateName).then(
+        (scenario) => {
+          dispatch({ type: "RECEIVE_SCENARIO", payload: scenario || null });
+        },
+      );
+    }
+  }, [state.stateName, state.sharedScenarios]);
+
+  useEffect(() => {
+    const systemType = "State Prison";
+
+    async function fetchFacilities(
+      scenario: Scenario | null,
+      stateName: string | null,
+    ) {
+      if (scenario && scenario.useReferenceData && stateName) {
+        let facilities = await fetchUserFacilities(scenario.id);
+        const referenceFacilities = await fetchReferenceFacilities(
+          stateName,
+          systemType,
+        );
+
+        facilities = buildCompositeFacilities(
+          facilities,
+          referenceFacilities,
+          scenario[referenceFacilitiesProp],
+        );
+        dispatch({
+          type: "RECEIVE_FACILITIES",
+          payload: Object.values(facilities),
+        });
+      } else {
+        if (stateName) {
+          const referenceFacilities = await fetchReferenceFacilities(
+            stateName,
+            systemType,
+          );
```

Comment:
You can probably move this `fetchReferenceFacilities` call outside of the `if/else` statement since we need to call it either way if `stateName` is present.  You may have to assign `referenceFacilities` to an empty array by default so that it can be in scope inside of the `if/else` statement branches